app/models/crate.js

```diff
@@ -67,6 +67,14 @@ export default class Crate extends Model {
     return Object.fromEntries(versions.slice().map(v => [v.id, v]));
   }
 
+  /** @return {Map<string, import("../models/version").default>} */
+  @cached
+  get loadedVersionsByNum() {
+    let versionsRef = this.hasMany('versions');
+    let values = versionsRef.value();
+    return new Map(values?.map(ref => [ref.num, ref]));
+  }
```

Comment:
I think `versionsObj` could also be changed to implement in this way. This would allow all `versions` that belong to this `crate` to be available, rather than just those loaded via the `loadVersionsTask`.